api/v1/invitations/PKPInvitationController.php

```diff
@@ -0,0 +1,236 @@
+     */
+    public function authorize(PKPRequest $request, array &$args, array $roleAssignments): bool
+    {
+        $illuminateRequest = $args[0]; /** @var \Illuminate\Http\Request $illuminateRequest */
+        $actionName = static::getRouteActionName($illuminateRequest);
+
+        $invitation = null;
+
+        $invitationType = $this->getParameter(self::PARAM_TYPE);
+        $invitationId = (int) $this->getParameter(self::PARAM_ID);
+        $invitationKey = $this->getParameter(self::PARAM_KEY);
+
+        if (in_array($actionName, $this->requiresType)) {
+            if (!isset($invitationType)) {
+                throw new Exception("Invitation Type parameter with the name '" . self::PARAM_TYPE . "' need to be declared");
+            }
+
+            $invitation = app(PKPInvitation::class)->create($invitationType);
+        } elseif (in_array($actionName, $this->requiresOnlyId)) {
+            if (!isset($invitationId)) {
+                throw new Exception("Invitation Id parameter with the name '" . self::PARAM_ID . "' need to be declared");
+            }
+
+            $invitationModel = Invitation::find($invitationId);
+            if (!isset($invitationModel)) {
+                throw new Exception("Invitation not found");
+            }
+
+            $invitation = app(PKPInvitation::class)->create($invitationModel->type, $invitationModel);
+        } elseif (in_array($actionName, $this->requiresIdAndKey)) {
+            if (!isset($invitationId) || !isset($invitationKey)) {
+                throw new Exception("Invitation Id parameter with the name '" . self::PARAM_ID . "' and invitation Key parameter with the name '" . self::PARAM_KEY . "' need to be declared");
+            }
+
+            $invitation = Repo::invitation()->getByIdAndKey($invitationId, $invitationKey);
+        }
+
+        if (!isset($invitation)) {
+            throw new Exception("Invitation could not be created");
+        }
+
+        $this->invitation = $invitation;
+
+        if (!array_key_exists(IApiHandleable::class, class_implements($this->invitation))) {
+            throw new Exception('This inviatation does not support API handling');
+        }
+
+        $this->createInvitationHandler = $invitation->getCreateInvitationController();
+        $this->receiveInvitationHandler = $invitation->getReceiveInvitationController();
+
+        if (in_array($actionName, $this->requiresType) || in_array($actionName, $this->requiresOnlyId)) {
+            $this->createInvitationHandler->authorize($this, $request, $args, $roleAssignments);
+        } elseif (in_array($actionName, $this->requiresIdAndKey)) {
+            $this->receiveInvitationHandler->authorize($this, $request, $args, $roleAssignments);
+        }
+
+        return parent::authorize($request, $args, $roleAssignments);
+    }
+
+    public function add(Request $illuminateRequest): JsonResponse
+    {
+        return $this->createInvitationHandler->add($illuminateRequest);
```

Comment:
I already have this code in that authorization:

```
if (!array_key_exists(IApiHandleable::class, class_implements($this->invitation))) {
   throw new Exception('This invitation does not support API handling');
}
```

In principle it will not go through the execution of the end point if it is not `IApiHandleable::class`. Also it will fail, again, in the authorize code.

I changed the `getCreateInvitationController` and `getReceiveInvitationController` functions to return a handler (instead of a nullable handler).

On top of that I have added a check in the authorization function like:

```
if (!isset($this->createInvitationHandler) || !isset($this->receiveInvitationHandler)) {
   throw new Exception('This invitation should have defined its API handling code');
}
```